doc: _extensions: kconfig: add :kconfig:option-regex: role

kartben opened the pull request:
Allow users to create links to Kconfig regex searches. The new role generates links to the Kconfig search page with the regex pattern as the search query.

Fixes zephyrproject-rtos/zephyr#90571

doc for the new role and demo of it in action: https://builds.zephyrproject.io/zephyr/pr/90681/docs/contribute/documentation/guidelines.html#role-kconfig-option-regex

<img width="854" alt="image" src="https://github.com/user-attachments/assets/b480eb27-82c0-43a2-acad-e3289bfad4ac" />

kartben committed:
doc: _extensions: kconfig: add :kconfig:option-regex: role

Allow users to create links to Kconfig regex searches.
The new role generates links to the Kconfig search page with the regex
pattern as the search query.

Fixes zephyrproject-rtos/zephyr#90571

Signed-off-by: Benjamin Cabé <benjamin@zephyrproject.org>

tomi-font reviewed (changes requested):
Thank you for implementing this! Didn't check the code, but the documentation looks (almost) good.
Would you mind at the same time updating the following to use the new role:
https://github.com/zephyrproject-rtos/zephyr/blob/3e8ec3aaf299ce1a59217a1321d8f966effa5da6/doc/services/storage/secure_storage/index.rst?plain=1#L79
https://github.com/zephyrproject-rtos/zephyr/blob/3e8ec3aaf299ce1a59217a1321d8f966effa5da6/doc/services/storage/secure_storage/index.rst?plain=1#L103

kartben committed:
doc: secure_storage: update Kconfig options formatting

Revised the documentation for PSA Secure Storage API configuration to
use the new :kconfig:option-regex: role for Kconfig options.

Signed-off-by: Benjamin Cabé <benjamin@zephyrproject.org>

tomi-font reviewed (approved):
:heart:
